specification/cognitiveservices/ContentSafetyStudio/main.tsp

```diff
@@ -0,0 +1,538 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+
+@service({
+  title: "Content Safety Studio API",
+})
+@versioned(ContentSafetyStudio.Versions)
+@useAuth(ApiKeyAuth<ApiKeyLocation.header, "Ocp-Apim-Subscription-Key">)
+@server(
+  "{endpoint}/contentsafety",
+  "Show metrics of content safety service",
+  {
+    @doc("""
+Supported Cognitive Services endpoints (protocol and hostname, for example:
+https://<resource-name>.cognitiveservices.azure.com).
+""")
+    endpoint: string,
+  }
+)
+@doc("Content Safety studio API for different metrics")
+namespace ContentSafetyStudio;
+
+enum Versions {
+  @useDependency(Azure.Core.Versions.v1_0_Preview_1)
+  `2023-04-30-preview`,
+}
+
+alias ServiceTraits = SupportsRepeatableRequests &
+SupportsConditionalRequests &
+SupportsClientRequestId;
+
+
+// Models ////////////////////
+@doc("Time granularity.")
+enum Granularity {
+  Hourly,
+  Daily
+}
+
+@doc("Modality type.")
+enum Modality {
+  Text,
+  Image
+}
+
+@doc("Content safety Operation.")
+enum ContentSafetyOperation {
+  AnalyzeText,
+  AnalyzeImage,
+}
+
+@doc("Content safety API version.")
+enum ContentSafetyVersion {
+  `2023-04-30-preview`
+}
+
+@doc("Category type.")
+enum CategoryEnum {
+  "Hate",
+  "Sexual",
+  "Violence",
+  "Self-harm" 
+}
+
+@doc("Severity level. The higher the value, the more risk it is.")
+enum SeverityEnum {
+  Zero: 0,
+  Two: 2,
+  Four: 4,
+  Six: 6
+}
+
+@doc("Aggregation methods for latency.")
+enum Aggregation {
+  P99,
+  P95,
+  Avg
+}
+
+
+@doc("API request count at a timestamp.")
+@resource("requestCounts")
+model CountPerTimestamp {
+  #suppress "@azure-tools/typespec-azure-core/key-visibility-required" "Properties in response-only schemas should not be marked readOnly"
+  @key("dateTime")
+  @doc("The date time.")
+  dateTime: utcDateTime;
```

Comment:
I am not sure if it is best solution to make it `resource` and have `key` on `dateTime`. This `CountPerTimestamp` or `LatencyPerTimestamp` seems not a resource. I assume it only for the `ResourceList` operation?
<-- ignore this comment, if this is recommended from TypeSpec team

Also check [Recommended Naming Conventions](https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md#recommended-naming-conventions) -- this is only a recommend.